src/app/containers/PageHandlers/withData.jsx

```diff
@@ -1,36 +1,55 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import { shape, element } from 'prop-types';
 import articlePropTypes from '../../models/propTypes/article';
 import ErrorMain from '../ErrorMain';
-import nodeLogger from '../../helpers/logger.node';
 import deepGet from '../../helpers/json/deepGet';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import getPassportHome from '../../lib/utilities/getPassportHome';
 
-const logger = nodeLogger(__filename);
-
-// checks for data and status keys, setting default status if not found
+// checks for data, status, setting default status if not found
 const constructRenderObject = data => ({
   status: deepGet(['status'], data) || 500,
   pageData: deepGet(['pageData'], data),
 });
 
-// checks for pageData and 200 status
+const validatePassportHome = passportHome => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { service } = useContext(ServiceContext);
+
+  return passportHome ? passportHome === service : true;
+};
+
+// checks for pageData, 200 status and if home service from article data fits the service locale
 const shouldRender = data => {
   const { status, pageData } = constructRenderObject(data);
 
+  let statusCode = status;
+  let isCorrectService;
+
   const hasDataAnd200Status = pageData && status === 200;
+  if (hasDataAnd200Status) {
+    const passportHome = getPassportHome(pageData);
+    isCorrectService = validatePassportHome(passportHome);
+    statusCode = !isCorrectService ? 404 : status;
+  }
 
-  return { hasDataAnd200Status, status, pageData };
+  return {
+    hasData200StatusAndCorrectService: hasDataAnd200Status && isCorrectService,
+    status: statusCode,
+    pageData,
+  };
 };
 
 const WithData = Component => {
   const DataContainer = ({ data, ...props }) => {
-    const { hasDataAnd200Status, status, pageData } = shouldRender(data);
-    if (hasDataAnd200Status) {
-      try {
```

Comment:
What's the reason for removing the try/catch block here?

Reply:
@sareh I removed it when I was bumping up the coverage for the file because I couldn't find a case where it caught errors for anything and I think we have an error boundary defined somewhere else (not sure of where yet though)